keps/sig-network/3698-multi-network-requirements/README.md

```diff
@@ -0,0 +1,352 @@
+    - [Story #9](#story-9)
+  - [Requirements](#requirements)
+    - [Phase I (base API and reference in Pod)](#phase-i-base-api-and-reference-in-pod)
+    - [Phase II (scheduler, kubelet and API probing for PodNetwork)](#phase-ii-scheduler-kubelet-and-api-probing-for-podnetwork)
+    - [Phase III (basic Kubernetes features integration)](#phase-iii-basic-kubernetes-features-integration)
+    - [Phase IV (extended functionality)](#phase-iv-extended-functionality)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Alpha](#alpha)
+    - [Beta](#beta)
+- [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+  - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+  - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+  - [Monitoring Requirements](#monitoring-requirements)
+  - [Dependencies](#dependencies)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+One of the main requirements for Kubernetes networking today is to enable
+connectivity between `Pods` in a cluster. This facility satisfies a large number of
+Kubernetes users, but it isn't sufficient for some important cases:
+
+* applications leveraging different isolated networks exposed through different
+  interfaces
+* applications leveraging performance-oriented interfaces (e.g. `AF_XDP`,
+  `memif`, `SR-IOV`)
+* applications requiring support for protocols not yet supported by Kubernetes
+
+Here we propose a solution that enables multiple network interfaces to be
+defined in the `Pod` specification so as to provide a standard and clear 
+mechanism to implement these more complex configurations.
+
+## Motivation
+
+We want to have a common API allowing us to define a catalog of different
+networks in the Kubernetes cluster. It would allow attaching a pod to one or
+several networks via a given type of interface depending on its connectivity
+or performance needs.
+
+### Goals
+
+Define user stories and requirements for the Multi-Network effort in Kubernetes.
+
+### Non-Goals
+
+Define API and implementation.
+
+## Proposal
+
+### Personas
+
+#### Cluster operator
+
+This role will be responsible for defining and managing PodNetworks (see below)
+that properly describe the infrastructure available for the cluster, namespace
+and workloads. This persona can define which users can “attach” to a specific
+PodNetwork.
+
+#### Application developer
+
+**Application developer** is the consumer of PodNetwork via referencing them in
+their workloads. Application developers usually will not create or remove the
+PodNetwork on their own.
+
+### Terminology
+
+* **PodNetwork** - name of the API object that will represent a network in
+  Kubernetes cluster
+* **Cluster Default PodNetwork** - This is the initial cluster-wide PodNetwork
```

Comment:
I think we should remove "cluster creation" and life cycle as this could change after create?

Suggest shorten to:

This is the PodNetwork attached to a Pod when no additional...

Reply:
done